### PR TITLE
Support MongoDB integration by nuget

### LINKED ISSUE #2328: NuGet package missing assemblies for MongoDB instrumentation
The package required for MongoDB bytecode instrumentation, [`MongoDB.Driver.Core.Extensions.DiagnosticSources`](https://www.nuget.org/packages/MongoDB.Driver.Core.Extensions.DiagnosticSources/), is only added via AdditionalDeps so it is not shipped with the current NuGet package.

cc @RassK

### PR DESCRIPTION
**Pull request by Kielek**
## Why

Fixes #2328

## What

Support MongoDB integration by nuget

## Tests

CI + ported test for one  mongodv to nuget tests.

## Checklist

- ~~[ ] `CHANGELOG.md` is updated.~~
- ~~[ ] Documentation is updated.~~
- ~~[ ] New features are covered by tests.~~

### CODE REVIEW

File: nuget/OpenTelemetry.AutoInstrumentation.Dependencies/OpenTelemetry.AutoInstrumentation.Dependencies.nuspec [hunk 1]
```diff
@@ -18,6 +18,7 @@
     <dependencies>
       <!-- TODO: Need a test/validation to make sure that this are in sync with actual projects -->
       <group targetFramework="net6.0">
+        <dependency id="MongoDB.Driver.Core.Extensions.DiagnosticSources" version="1.3.0" />
```
Comment (pjanotti) [inline nuget/OpenTelemetry.AutoInstrumentation.Dependencies/OpenTelemetry.AutoInstrumentation.Dependencies.nuspec:21]: The package should be added to the `OpenTelemetry.AutoInstrumentation` project instead. There is some overlap with the dependencies package, and we may end up dropping one of them. Anyway, for now the right place for this is on `OpenTelemetry.AutoInstrumentation`.

**Review by pjanotti — COMMENTED** [review]
Nice way to re-use the test app and helpers!

File: nuget/OpenTelemetry.AutoInstrumentation.Dependencies/OpenTelemetry.AutoInstrumentation.Dependencies.nuspec (hunk 1 above)
Comment (RassK) [reply nuget/OpenTelemetry.AutoInstrumentation.Dependencies/OpenTelemetry.AutoInstrumentation.Dependencies.nuspec:21]: Adding it to `OpenTelemetry.AutoInstrumentation` will make the dependencies shipping even worse. Nulling all optimizations like https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/2294

Comment (pjanotti) [reply nuget/OpenTelemetry.AutoInstrumentation.Dependencies/OpenTelemetry.AutoInstrumentation.Dependencies.nuspec:21]: That is true @RassK I think the path to addressing that is via #2329 - right now the NuGet package is broken for MongoDB.

Comment (Kielek) [reply nuget/OpenTelemetry.AutoInstrumentation.Dependencies/OpenTelemetry.AutoInstrumentation.Dependencies.nuspec:21]: I am checking second option locally:
 - move MogoDB.Instrumentation as a dependency from Additional Store directly to instrumentation (as side effect .NET Fx should work)
 - OpenTelemetry.AutoInstrumentation will be referencing MongoDb.Instrumentation.

Comment (RassK) [reply nuget/OpenTelemetry.AutoInstrumentation.Dependencies/OpenTelemetry.AutoInstrumentation.Dependencies.nuspec:21]: > @RassK, we don't need to add it to `OpenTelemetry.AutoInstrumentation` __csproj__ we can temporarily have it directly on the `OpenTelemetry.AutoInstrumentation` __nuspec__. 

I think this is the only reasonable idea for now. This is one of the worst direct references to take for our project.

Comment (pjanotti) [reply nuget/OpenTelemetry.AutoInstrumentation.Dependencies/OpenTelemetry.AutoInstrumentation.Dependencies.nuspec:21]: See discussion at https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/2397#issuecomment-1505423109

**Pull request closed without merging**